kernel/src/process.rs

```diff
@@ -219,8 +219,13 @@ pub trait ProcessType {
     /// Context switch to a specific process.
     unsafe fn switch_to(&self) -> Option<syscall::ContextSwitchReason>;
 
-    unsafe fn fault_fmt(&self, writer: &mut dyn Write);
-    unsafe fn process_detail_fmt(&self, writer: &mut dyn Write);
+    /// Print out the memory map (Grant region, heap, stack, program
+    /// memory, BSS, and data sections) of this process.
+    unsafe fn print_memory_map(&self, writer: &mut dyn Write);
+
+    /// Print out the full state of the process: its memory map, its
+    /// context, and the state of the memory protection unit (MPU).
+    unsafe fn print_full_process(&self, writer: &mut dyn Write);
```

Comment:
It's great to have more of a clear intent behind these functions.

Reply:
`print_full_process` is simply a renaming of `process_detail_fmt`. `print_memory_map` factors out the part of this function that's actually implemented in `Process`. That way you can choose to print the memory map without also having to print all of the other state.